packages/components/src/badge/index.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { info, caution, error, published } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import type { BadgeProps } from './types';
+import Icon from '../icon';
+
+function Badge( {
+	className,
+	as: Component = 'div',
+	context = 'neutral',
+	children,
+	...props
+}: BadgeProps ) {
+	/**
+	 * Returns an icon based on the badge context.
+	 *
+	 * @return {JSX.Element | null} The corresponding icon for the provided context.
```

Comment:
The type in the JSDoc can be omitted because the actual type annotation in the TS makes it redundant.

```suggestion
	 * @return The corresponding icon for the provided context.
```